### PR TITLE
Reformats the hosts table to use consistent units and sorting

### PR DESCRIPTION
**Pull request by mtlynch**
- Updates the hosts page of Sia-Explorer so that host attributes are all displayed in consistent units.
- Sorts the hosts by address, storage capacity, or price and allows user to dynamically change sorting.

### CODE REVIEW

**Comment by lukechampine** [comment]
I'm not sure how I feel about adding JS to the explorer. We're going to give it a big makeover soon (mostly ~~improving~~ adding styling to match our new design language). At that point we can reevaluate whether JavaScript is necessary.
For now I'd prefer to keep things simple. Unit conversion and sorting can be done server-side, although we'd need to add a query param to dictate what to sort by.

**Comment by mtlynch** [comment]
Darn, I guess I should have talked to you guys first. I assumed lack of JavaScript was just because you wanted something quick 'n dirty.

**Comment by DavidVorick** [comment]
@lukechampine switching to javascript was the plan the whole time, there's a lot more flexibility for things like adding fancy graphs and making it very visually appealing. Aaron is probably going to be largely rewriting the explorer from scratch.

This PR is worth a second look.

**Comment by mtlynch** [comment]
Reopening per @DavidVorick's comment

File: hosts.go [hunk 1]
```diff
@@ -4,39 +4,49 @@ import (
...
-type hostList struct {
-	Hosts  []modules.HostSettings
-	Length int
+// hostDisplayInfo contains the data about each host that will be displayed on
+// the hosts page.
+type hostDisplayInfo struct {
+	IPAddress    modules.NetAddress
+	TotalStorage int64
+	Price        types.Currency
+}
```
Comment (lukechampine) [inline hosts.go:18]: what is the purpose of this type? Just to restrict the available information? Doesn't seem worth the hassle.

File: static/hosts.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+	// Populates the table of known hosts.
+	function populateHostTable() {
+		$('#hostsTable > tbody').empty();
+		var hosts = $.hosts.hosts;
+		sortFunc = getSortFunc();
+		hosts.sort(sortFunc);
+		hosts.forEach(function (host) {
+			var storageFormatted = formatStorage(host.TotalStorage, "GB");
+			var priceFormatted = (host.Price / 1000000).toFixed(2) + " SC";
```
Comment (lukechampine) [inline static/hosts.js]: why is the conversion factor 1000000?

File: static/hosts.html [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+				<td>Number of Hosts</td>
+				<td id="hostCount"></td>
+      </tr>
+    </table>
+    <table id="hostsTable">
+			<thead>
+				<tr>
+					<th id="addrHeader"><a href="#">Host Address</a></th>
+					<th id="storageHeader"><a href="#">Host Storage</a></th>
+					<th id="priceHeader"><a href="#">Price (per block per month)</a></th>
```
Comment (lukechampine) [inline static/hosts.html:22]: block and month are both units of time. Maybe you meant GB/month?

File: hosts.go (hunk 1 above)
Comment (mtlynch) [inline hosts.go:18]: This is the type that gets serialized and published to users when they query `/hosts.json`. I think there's value in limiting what we output in the spirit of ["Be conservative in what you send, be liberal in what you accept"](https://en.wikipedia.org/wiki/Robustness_principle).

File: static/hosts.html (hunk 3 above)
Comment (mtlynch) [inline static/hosts.html:22]: Ah, I only got that from snippets of conversation in slack. I thought files were broken up into "blocks" (or chunks) and the price was per block (chunk) of data per month, but I now understand "block" refers to blockchain block.

What are the units on `HostSettings.Price`? Is it price per GB/month?

File: static/hosts.js (hunk 2 above)
Comment (mtlynch) [inline static/hosts.js]: Whoops, I thought it was 1 million H = 1 SC. I was [way off](http://stream1.gifsoup.com/view2/20140916/5109205/samsonite-i-was-way-off-o.gif).

Fixed!

File: explorer.go [hunk 4]
```diff
@@ -109,11 +109,15 @@ func main() {
 		serveMux: http.NewServeMux(),
 	}
 
-	es.serveMux.Handle("/", http.FileServer(http.Dir("./webroot/")))
+	http.Handle("/static/", http.StripPrefix("/static", http.FileServer(http.Dir("./static/"))))
```
Comment (DavidVorick) [inline explorer.go:112]: what is up with this?

**Comment by DavidVorick** [comment]
Unfortunately buried. We're going to be cleaning up the explorer at some point, but has fallen pretty far off the priority list for the time being.

**Pull request closed without merging**